### PR TITLE
Feature-requirements-for-returns-view-page

### PR DESCRIPTION
**Pull request by jonathangoulding**
https://eaflood.atlassian.net/browse/WATER-4422

The requirements for return journey has been built and the check page has been finished. This showed a page with requirements for returns allowing users to make changes to requirements for returns.

This piece of work will replicate the requirements for returns check page but will not allow users to make changes (the card and information is identical, apart from the title and a return reference).

There will be a link "View" on the licence set up page that links to this page. That is added as part of this change. Work for this was done here - https://github.com/DEFRA/water-abstraction-system/pull/1054

### CODE REVIEW

File: app/lib/general.lib.js [hunk 1]
```diff
@@ -152,6 +152,36 @@ function generateAbstractionPointDetail (pointDetail) {
   return abstractionPoint
 }
 
+/**
+ * Generate a string that represents an abstraction point based on the new data (obs change this in PR)
+ *
+ */
+function generatePointDetail (point) {
+  let abstractionPoint = null
```
Comment (jonathangoulding) [inline app/lib/general.lib.js:160]: Add test for this

File: app/views/return-requirements/view.njk [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    <hr class="govuk-section-break  govuk-!-margin-bottom-2 govuk-section-break--visible">
+  </div>
+
+{% if notes %}
+  <div class="govuk-!-margin-bottom-9">
+    <h2 class="govuk-heading-l govuk-!-margin-bottom-4">Notes</h2>
+    <hr class="govuk-section-break  govuk-!-margin-bottom-2 govuk-section-break--visible">
+    {{ govukSummaryList({
+      classes: 'govuk-!-margin-bottom-2',
+      rows: [
```
Comment (jonathangoulding) [inline app/views/return-requirements/view.njk:64]: Currently awaiting a decision on the user email. 

Currently this is not stored nor does it exists in the legacy data. As this affect many records a redesign might be required.

File: app/lib/general.lib.js [hunk 3]
```diff
@@ -152,6 +152,45 @@ function generateAbstractionPointDetail (pointDetail) {
...
+ * Nation Grid References are saved against the abstraction point. This function checks for these references and builds
+ * a string that defines the details of the abstraction point.
+ *
+ * This follows the same out put as generateAbstractionPointDetail but the points have already been merged
+ *
+ * @param {Object} pointDetail - Object containing all the details for the point
+ *
+ * @returns {String} a description of the abstraction point
+ */
+function generatePointDetail (pointDetail) {
```
Comment (jonathangoulding) [inline app/lib/general.lib.js:168]: Might need to work on this name and docs.

My current understanding is limiting me to explain it better.

File: app/views/return-requirements/check.njk [hunk 4]
```diff
@@ -75,34 +75,20 @@
...
-                href: "delete-note",
-                text: "Delete"
-              } if note
-            ]
-          }
-        }
-      ]
-    }) }}
+    <div class="govuk-summary-list govuk-!-margin-bottom-1">
+      <p class="govuk-body">{{ note.text | escape }}</p>
```
Comment (jonathangoulding) [inline app/views/return-requirements/check.njk:79]: Old logic was - ` text: note | escape | nl2br`

Is this still needed ?

**Review by Cruikshanks — APPROVED** [review]

**Pull request merged**